<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>776</width>
    <height>636</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form1</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Дані</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="1" column="1">
       <widget class="QToolButton" name="select_input">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLineEdit" name="line_output">
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>Вихідний файл</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QToolButton" name="select_output">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="line_input">
        <property name="text">
         <string/>
        </property>
        <property name="placeholderText">
         <string>Вхідний файл</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" rowspan="2">
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Поліноми</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Типи</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QRadioButton" name="radio_sh_cheb">
           <property name="text">
            <string>Комбінований</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_cheb">
           <property name="text">
            <string>Лаґерра</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radio_sh_cheb_2">
           <property name="text">
            <string>Зміщений Чеб.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="custom_check">
           <property name="text">
            <string>Власна структура</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Степені</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4" columnstretch="1,0">
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>X1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="x3_deg">
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>X3</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="x2_deg">
           <property name="value">
            <number>6</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>X2</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="x1_deg">
           <property name="value">
            <number>7</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QPushButton" name="bruteforce_btn">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Підібрати</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Розмірність</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>X3</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="x1_dim">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="value">
            <number>7</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>X1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="y_dim">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>X2</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="x2_dim">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="value">
            <number>6</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="x3_dim">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" rowspan="2">
    <widget class="QGroupBox" name="groupBox_6">
     <property name="title">
      <string>Обробка</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="3" column="2">
       <widget class="QSpinBox" name="predictBox">
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Кроки передбачення:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Навчальна вибірка (N_02)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QSpinBox" name="sample_spin">
        <property name="maximum">
         <number>999</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0" rowspan="2" colspan="3">
       <widget class="QPushButton" name="exec_button">
        <property name="text">
         <string>Порахувати</string>
        </property>
        <property name="default">
         <bool>true</bool>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_7">
     <property name="title">
      <string>Додатково</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Ваги:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="weights_box">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>Average</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Scaled</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="lambda_check">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>3-блочні вирази</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QTableWidget" name="tablewidget">
     <property name="maximumSize">
      <size>
       <width>740</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="rowCount">
      <number>10</number>
     </property>
     <property name="columnCount">
      <number>8</number>
     </property>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <row/>
     <column>
      <property name="text">
       <string>№
виб</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Прибуток</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Запас
ходу</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Енергія
АБ</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Стан
фунціонування</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Ризик
аварії</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Причина
нештатної
ситуації</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Рівень
небезпеки</string>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string/>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
     </item>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>radio_sh_cheb</tabstop>
  <tabstop>radio_cheb</tabstop>
  <tabstop>radio_sh_cheb_2</tabstop>
  <tabstop>x1_deg</tabstop>
  <tabstop>x2_deg</tabstop>
  <tabstop>x3_deg</tabstop>
  <tabstop>x1_dim</tabstop>
  <tabstop>x2_dim</tabstop>
  <tabstop>x3_dim</tabstop>
  <tabstop>y_dim</tabstop>
  <tabstop>line_input</tabstop>
  <tabstop>select_input</tabstop>
  <tabstop>line_output</tabstop>
  <tabstop>select_output</tabstop>
  <tabstop>exec_button</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>exec_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>exec_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>379</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>select_input</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>input_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>select_output</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>output_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x1_dim</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>dimension_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x2_dim</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>dimension_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x3_dim</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>dimension_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x1_deg</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>degree_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x2_deg</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>degree_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x3_deg</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>degree_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>279</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radio_sh_cheb</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>type_modified(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radio_sh_cheb_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>type_modified(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radio_cheb</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>type_modified(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Form</sender>
   <signal>output_changed(QString)</signal>
   <receiver>line_output</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Form</sender>
   <signal>input_changed(QString)</signal>
   <receiver>line_input</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line_input</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Form</receiver>
   <slot>input_modified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>483</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line_output</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Form</receiver>
   <slot>output_modified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>483</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lambda_check</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>lambda_calc_method_changed(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>weights_box</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>Form</receiver>
   <slot>weights_modified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_dim</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>dimension_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>378</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bruteforce_btn</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>bruteforce_called()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>171</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>custom_check</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>structure_changed(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sample_spin</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>samples_modified(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>input_changed(QString)</signal>
  <signal>output_changed(QString)</signal>
  <signal>x1_dim_changed(int)</signal>
  <signal>x2_dim_changed(int)</signal>
  <signal>x3_dim_changed(int)</signal>
  <signal>x1_deg_changed(int)</signal>
  <signal>x2_deg_changed(int)</signal>
  <signal>x3_deg_changed(int)</signal>
  <signal>type_cheb()</signal>
  <signal>type_lege()</signal>
  <signal>type_lagg()</signal>
  <signal>type_herm()</signal>
  <slot>input_clicked()</slot>
  <slot>output_clicked()</slot>
  <slot>samples_modified(int)</slot>
  <slot>dimension_modified(int)</slot>
  <slot>degree_modified(int)</slot>
  <slot>type_modified(bool)</slot>
  <slot>plot_clicked()</slot>
  <slot>exec_clicked()</slot>
  <slot>input_modified(QString)</slot>
  <slot>output_modified(QString)</slot>
  <slot>weights_modified(QString)</slot>
  <slot>lambda_calc_method_changed(bool)</slot>
  <slot>bruteforce_called()</slot>
  <slot>structure_changed(bool)</slot>
 </slots>
</ui>
